plugins/sql-language/src/main/java/org/polypheny/db/sql/language/ddl/SqlDropTable.java

```diff
@@ -51,27 +48,19 @@ public class SqlDropTable extends SqlDropObject {
 
 
     @Override
-    public void execute( Context context, Statement statement, QueryParameters parameters ) {
-        final CatalogTable table;
+    public void execute( Context context, Statement statement, ParsedQueryContext parsedQueryContext ) {
+        final Optional<LogicalTable> table = searchEntity( context, name );
 
-        try {
-            table = getCatalogTable( context, name );
-        } catch ( PolyphenyDbContextException e ) {
+        if ( table.isEmpty() ) {
             if ( ifExists ) {
-                // It is ok that there is no database / schema / table with this name because "IF EXISTS" was specified
+                // It is ok that there is no table with this name because "IF EXISTS" was specified
                 return;
             } else {
-                throw e;
+                throw new GenericRuntimeException( "There exists no table with the name %s and 'IF EXISTS' was not specified", name );
             }
         }
 
-        try {
-            DdlManager.getInstance().dropTable( table, statement );
-        } catch ( DdlOnSourceException e ) {
-            throw CoreUtil.newContextException( name.getPos(), RESOURCE.ddlOnSourceTable() );
-        }
-
+        DdlManager.getInstance().dropTable( table.get(), statement );
```

Comment:
There is no check whether the entity found by searchEntity actually is a table and not another schema object like, for instance, a view.

Reply:
Added check for entity or source with appropriate error.